apps/microtvm/arduino/template_project/microtvm_api_server.py

```diff
@@ -313,7 +304,51 @@ def _find_modified_include_path(self, project_dir, file_path, include_path):
         # It's probably a standard C/C++ header
         return include_path
 
+    def _cmsis_required(self, project_path: Union[str, pathlib.Path]) -> bool:
+        """Check if CMSIS dependency is required."""
+        project_path = pathlib.Path(project_path)
+        for path in (project_path / "src" / "model").iterdir():
+            if path.is_file():
+                with open(path, "r", encoding="ISO-8859-1") as lib_f:
+                    lib_content = lib_f.read()
+                if any(
+                    header in lib_content
+                    for header in [
+                        "<arm_nnsupportfunctions.h>",
+                        "arm_nn_types.h",
+                        "arm_nnfunctions.h",
+                    ]
+                ):
```

Comment:
Could we store the CMSIS_NN headers as a capitalized constant elsewhere? While we're at it, let's also include [all five](https://github.com/ARM-software/CMSIS_5/tree/develop/CMSIS/NN/Include) of the headers, not just three, e.g.
```python
    CMSIS_INCLUDE_HEADERS = [
        "arm_nn_math_types.h",
        "arm_nn_tables.h",
        "arm_nn_types.h",
        "arm_nnfunctions.h",
        "arm_nnsupportfunctions.h",
    ]
```